src/browser/commands/saveState/index.ts

```diff
@@ -32,23 +31,10 @@ export const defaultOptions = {
     sessionStorage: true,
 };
 
-export const getCalculatedProtocol = (browser: Browser): typeof DEVTOOLS_PROTOCOL | typeof WEBDRIVER_PROTOCOL => {
-    const protocol = browser.config.automationProtocol;
-
-    if (protocol === DEVTOOLS_PROTOCOL) {
-        return DEVTOOLS_PROTOCOL;
-    }
-
-    if (
-        protocol === WEBDRIVER_PROTOCOL &&
-        browser.config.isolation &&
-        isSupportIsolation(browser.publicAPI.capabilities.browserName!, browser.publicAPI.capabilities.browserVersion!)
-    ) {
-        return DEVTOOLS_PROTOCOL;
-    }
-
-    return protocol;
-};
+export const getProtocol = (browser: Browser): typeof WEBDRIVER_PROTOCOL | typeof DEVTOOLS_PROTOCOL =>
+    browser.config.automationProtocol === WEBDRIVER_PROTOCOL && browser.publicAPI.isBidi && browser.config.isolation
+        ? DEVTOOLS_PROTOCOL
+        : browser.config.automationProtocol;
```

Comment:
Function is very confusing.
Lets rename it to, like, "shouldUsePuppeteer" with logic like:
```js
if (browser.config.automationProtocol === WEBDRIVER_PROTOCOL && browser.publicAPI.isBidi && browser.config.isolation) {
    return true;
}

return browser.config.automationProtocol === DEVTOOLS_PROTOCOL;
```

It can't be described, why "getProtocol" returns "devtools" when i am using "automationProtocol: webdriver"